src/renderer/dx/DxRenderer.cpp

```diff
@@ -833,6 +1013,16 @@ void DxEngine::_InvalidOr(RECT rc) noexcept
 
     _invalidScroll = { 0 };
 
+    if (_retroTerminalEffects)

```

Comment:
I'm not too clear on our Presentation phase, but should this be part of that? Down in `::Present`.

Reply:
I'm not sure how to search the resolved conversations but someone suggested to move this here instead of its own phase because then we'd need empty stubs in the other renderers.

Reply:
Oh, huh. I believe we already have a `Present` phase, though?

Reply:
Oh I forgot we kept the `Present` method. I honestly don't know which location would be more appropriate - `EndPaint` occurs under lock, while `Present` does not. @miniksa would _definitely_ know though

Reply:
So Present occurs outside of the lock on the buffer, which should be more appropriate here- we don’t want to hold up text input while postprocessing the frame for presentation.